datalad/support/gitrepo.py

```diff
@@ -3109,12 +3154,14 @@ def _diffstatus_get_state_props(self, f, from_state, to_state,
         Parameters
         ----------
         f : Path
-        from_state : dict
-        to_state : dict
+        from_state : dict or None
+        to_state : dict or None
         against_commit : bool
           Flag whether `to_state` reflects a commit or the worktree.
-        modified_in_worktree : bool
-          Flag whether a worktree modification is reported. This is ignored
+        modified_in_worktree : False or str
+          False if there is no modification of `f` in the working tree,
+          or a type label from GIT_MODE_TYPE_MAP indicating the type
+          of the modified path `f`. This is ignored
```

Comment:
```suggestion
          of the modified path `f`. This is ignored. Could be `None` to signal that file was deleted.
```
correct (judging from the [code below](https://github.com/datalad/datalad/pull/6797/files#diff-bfc1c47d032fc72b15b6a79db3d49e4c4334cf96318c5efe48a3403744b32fb1R3214)?